ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { renderComponent } from "../../../components/__tests__/ForceScrollDownView.test";
+import * as profileBannerImport from "../../../features/profileSettings/store/selectors";
+import { GlobalState } from "../../../store/reducers/types";
+import { ProfileMainScreenTopBanner } from "../ProfileMainScreenTopBanner";
+
+const mockNavigate = jest.fn();
+jest.mock("@react-navigation/native", () => ({
+  ...jest.requireActual<typeof import("@react-navigation/native")>(
+    "@react-navigation/native"
+  ),
+  useNavigation: () => ({
+    navigate: mockNavigate
+  })
+}));
+
+describe("ProfileMainScreenTopBanner", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it("should match snapshot for all possible results of profileBannerToShowSelector", () => {
+    const testCases = [
+      {
+        selectorValue: "NOTIFICATIONS",
+        description: "notifications banner"
+      },
+      {
+        selectorValue: "PROFILE_BANNER",
+        description: "profile banner"
+      },
+      {
+        selectorValue: undefined,
+        description: "no banner"
+      }
+    ] as const;
+    testCases.forEach(({ selectorValue, description }) => {
+      jest
+        .spyOn(profileBannerImport, "profileBannerToShowSelector")
+        .mockImplementation((_: GlobalState) => selectorValue);
+
+      const component = renderComponent(<ProfileMainScreenTopBanner />);
+
+      expect(component).toMatchSnapshot(description);
```

Comment:
Why don't we do `expect(component.toJSON())`?

Reply:
didn't see the need -- added!